### ISSUE TITLE
"int not iterable" during presence timeout handling

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#3715]
Synapse: 0.33.2 with unrelated patches.

```
homeserver_1 - 2018-08-17 18:27:51,422 - synapse.handlers.presence - 345 - INFO - Handling presence timeouts
homeserver_1 - 2018-08-17 18:27:51,423 - synapse.handlers.presence - 386 - ERROR - Exception in _handle_timeouts loop
Traceback (most recent call last):
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/synapse/handlers/presence.py", line 364, in _handle_timeouts
    self.external_process_last_updated_ms.pop(process_id, ())
TypeError: 'int' object is not iterable
```

**Comment by turt2live** [comment]
Updated stacktrace:

```
homeserver_1 - 2018-10-02 16:15:02,799 - synapse.handlers.presence - 350 - INFO -  - Handling presence timeouts
homeserver_1 - 2018-10-02 16:15:02,799 - synapse.handlers.presence - 391 - ERROR -  - Exception in _handle_timeouts loop
Traceback (most recent call last):
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/synapse/handlers/presence.py", line 369, in _handle_timeouts
    self.external_process_last_updated_ms.pop(process_id, ())
TypeError: 'int' object is not iterable
```

**Comment by ulope** [comment]
Traceback with v1.3.1:

```
2019-09-26 00:38:53,827 synapse.metrics.background_process_metrics(202) [ERROR   ]: handle_presence_timeouts-189 Background process 'handle_presence_timeouts' threw an exception
Traceback (most recent call last):
  File "/synapse-venv/lib/python3.7/site-packages/synapse/metrics/background_process_metrics.py", line 200, in run
    yield func(*args, **kwargs)
  File "/synapse-venv/lib/python3.7/site-packages/synapse/handlers/presence.py", line 353, in _handle_timeouts
    self.external_process_last_updated_ms.pop(process_id, ())
TypeError: 'int' object is not iterable
```

**Comment by richvdh** [comment]
this code looks completely broken. I'm sure it's never worked. Unfortunately fixing it is going to require grokking what it is meant to do and writing some tests.

**Comment by ulope** [comment]
We're seeing those exceptions very frequently.

Recently we've also started to experience serious presence update issues (e.g. users on the other side of a federated server don't receive presence updates).

Could this be a cause for that?

**Comment by Dominik1999** [comment]
Can you help us with this issue? Raiden atm somehow relies on properly working presence states. And somehow since 1 month, this is not working anymore. 

See problem description here https://gist.github.com/ulope/1310338587ee7a794e550aca2a0c5d6d

We would be willing to pay for a quick solution to that problem if needed.

**Comment by neilisfragile** [comment]
@Dominik1999 @ulope 
We're working on it as a matter of urgency, will get a fix out as soon as we can, though as above, we don't consider it to be a trivial.

**Comment by richvdh** [comment]
hopefully fixed by #6212

**Comment by Dominik1999** [comment]
thanks again for fixing this!
